rpc/src/rpc.rs

```diff
@@ -5700,6 +5703,69 @@ pub mod tests {
         );
     }
 
+    #[test]
+    #[should_panic(expected = "should be less than 128 bytes")] // If ever `MAX_BASE58_BYTES` changes, the expected error message will need to be updated.
+    fn test_encode_account_throws_when_data_too_large_to_base58_encode() {
+        let data = vec![42; MAX_BASE58_BYTES + 1];
+        let pubkey = Pubkey::new_unique();
+        let account = AccountSharedData::create(42, data, pubkey, false, 0);
+        let _ = encode_account(&account, &pubkey, UiAccountEncoding::Base58, None).unwrap();
+    }
+
+    #[test]
+    fn test_encode_account_does_not_throw_despite_data_too_large_to_base58_encode_because_dataslice_makes_it_fit(
```

Comment:
Please make this and following test names more succinct. Feel free to add comments by the asserts if you think the logic needs this much explanation.

Reply:
Can do!

I would ask you to reconsider, though. When the name of the test _in the test runner logs_ tells the next person everything they need to know about what went wrong and why – without having to read the code of the test – then everyone can move a bit faster.

Consider your suggestion to remove `throws_when_*_to_base58_encode` from the test above: `test_encode_account_base58_data_too_large()`. Now the next person encounters this test failure: why did it fail? Because the data was too large? The inputs were too large? No exception was thrown, but the test failed. Now they're wondering why it failed if nothing went wrong. Into the source code they go…

All of that flailing is prevented when the test name is `test_encode_account_throws_when_data_too_large_to_base58_encode()`.

Reply:
I can't really imagine any case where a change would break a test and we wouldn't need to go look at the test anyway. My personal preference is for the test itself to be self-descriptive. That said, I don't care enough to argue this further.